### ISSUE TITLE
New endpoint that returns the transfers for a list of identities

### ISSUE THREAD

**Issue by sallymoc** [qubic/go-archiver#62]
- The endpoint can receive as input one or more identities
- The endpoint should return the list of transactions paginated, using cursor-based pagination (optional cursor parameter allow to handle the next pages).
- The following parameters can be also used to filter the list:
-- Direction (incoming, outgoing or any)
-- Status
-- Date from, date to

**Comment by LINCKODE** [comment]
Hey there!
Could you perhaps go in a bit more detail about the structure of the response?
I'm having a bit of a hard time understanding the exact response requirements for this endpoint.

**Comment by ahmed-tarek-salem** [comment]
Hey @LINCKODE,

Let me clarify:

- The endpoint will be updated to handle multiple identities. Instead of the current `/v2/identities/{identity}/transfers`, it should allow passing a list of public IDs as input and return their corresponding transfers.
  
- The response should be **paginated** using cursor-based pagination, with a cursor parameter for navigating between pages. You can find an example of this here 
https://github.com/qubic/go-archiver/issues/60#issuecomment-2369433093

- Additionally, the response should support **filtering** based on the following criteria:
  - **Direction:** incoming, outgoing, or any.
  - **Status**
  - **Date range:** `from` and `to`.

- By default, the results should be **sorted by date in descending order**.

Please let me know if you need further details.

**Comment by sallymoc** [comment]
Please @LINCKODE also ensure that transfers generated from sendmany procedure execution are also included.
i.e 
From MPIPDLVJKZKTCCODDCYNQXRAYJXBYXFJERPOHWJEGFIQLPZAMMKSMCIABMKC, 3 transfers where done in the transaction below, tick 17,393,385, to WHWSLPEYRPMKFBVQUQAMLCKHYAQDXHJKRVMVQVLLBCOHBRIWTTUWQODDVDXG
https://explorer.qubic.org/network/tx/bnklqwjqpkdofazhetkoqfnytkxfehllhtlexascyeuqlrnjhgzbucgbwdea?type=latest

However, if I navigate to the explorer identity page, I do not see them (trx are sorted by tick) :

![image](https://github.com/user-attachments/assets/45b4aaaf-c9b9-4ba0-b529-15b43477293e)


https://explorer.qubic.org/network/address/WHWSLPEYRPMKFBVQUQAMLCKHYAQDXHJKRVMVQVLLBCOHBRIWTTUWQODDVDXG



This is because the transfer service is not returning it in the results.

**Comment by 0xluk** [comment]
@sallymoc `transfers generated from sendmany procedure execution` this will be part of another service which is currently in development

**Comment by sallymoc** [comment]
Because in the wallets, we can see a list of the transactions. A wallet can have multiple accounts. 
Doesn't make sense that we make a call per account, we can't paginate the results, we would be making many calls every time we need to fetch the data.
The existing service the wallet app is using, is provided by qubic.li and we need to replace it by qubic.org services.

**Comment by sallymoc** [comment]
> @sallymoc `transfers generated from sendmany procedure execution` this will be part of another service which is currently in development

@0xluk  Could you please share the issue/specs you have for this?
What does this mean? We need to call two services to get the list of transfers?

**Comment by 0xluk** [comment]
> > @sallymoc `transfers generated from sendmany procedure execution` this will be part of another service which is currently in development
> 
> @0xluk Could you please share the issue/specs you have for this? What does this mean? We need to call two services to get the list of transfers?

No, this new service will expose all transfers (triggered by a normal transfer tx or a smart contract call tx)

**Comment by sallymoc** [comment]
This is blocking the issue https://github.com/qubic/wallet-app/issues/210

**Comment by 0xluk** [comment]
Due to recent development progress, this can be part of the new go-transfers service; will add it to the board to decide priority
